test/testactivitylistmodel.cpp

```diff
@@ -0,0 +1,416 @@
+    {
+        fakeQnam.reset(new FakeQNAM({}));
+        account = OCC::Account::create();
+        account->setCredentials(new FakeCredentials{fakeQnam.data()});
+        account->setUrl(QUrl(("http://example.de")));
+
+        accountState.reset(new OCC::AccountState(account));
+
+        fakeQnam->setOverride([this](QNetworkAccessManager::Operation op, const QNetworkRequest &req, QIODevice *device) {
+            Q_UNUSED(device);
+            QNetworkReply *reply = nullptr;
+
+            const auto urlQuery = QUrlQuery(req.url());
+            const auto format = urlQuery.queryItemValue(QStringLiteral("format"));
+            const auto since = urlQuery.queryItemValue(QStringLiteral("since")).toInt();
+            const auto limit = urlQuery.queryItemValue(QStringLiteral("limit")).toInt();
+            const auto path = req.url().path();
+
+            if (!req.url().toString().startsWith(accountState->account()->url().toString())) {
+                reply = new FakeErrorReply(op, req, this, 404, fake404Response);
+            }
+            if (format != QStringLiteral("json")) {
+                reply = new FakeErrorReply(op, req, this, 400, fake400Response);
+            }
+
+            if (path.startsWith(QStringLiteral("/ocs/v2.php/apps/activity/api/v2/activity"))) {
+                reply = new FakePayloadReply(op, req, FakeRemoteActivityStorage::instance()->activityJsonData(since, limit), searchResultsReplyDelay, fakeQnam.data());
+            }
+
+            if (!reply) {
+                return qobject_cast<QNetworkReply*>(new FakeErrorReply(op, req, this, 404, QByteArrayLiteral("{error: \"Not found!\"}")));
+            }
+
+            return reply;
+        });
+
+        OCC::AccountManager::instance()->addAccount(account);
+
+        // Activity comparison is done by checking type, id, and accName
+        // We need an activity with these details, at least
+        testNotificationActivity._accName = accountState->account()->displayName();
+        testNotificationActivity._id = 1;
+        testNotificationActivity._type = OCC::Activity::NotificationType;
+        testNotificationActivity._dateTime = QDateTime::currentDateTime();
+    };
+
+    // Test receiving activity from server
+    void testFetchingRemoteActivity() {
+        TestingALM model;
+        model.setAccountState(accountState.data());
+        QScopedPointer<QAbstractItemModelTester> modelTester(new QAbstractItemModelTester(&model));
+
+        QCOMPARE(model.rowCount(), 0);
+
+        model.startFetchJob();
+        QSignalSpy activitiesJob(&model, &TestingALM::activityJobStatusCode);
+        QVERIFY(activitiesJob.wait(3000));
+        QCOMPARE(model.rowCount(), 50);
+    };
+
+    // Test receiving activity from local user action
+    void testLocalSyncFileAction() {
+        TestingALM model;
+        model.setAccountState(accountState.data());
+        QScopedPointer<QAbstractItemModelTester> modelTester(new QAbstractItemModelTester(&model));
```

Comment:
you do need a pointer and can directly use `QAbstractItemModelTester modelTester(&model);`

Reply:
Ah, of course - replaced :)